### ISSUE TITLE
[css-text-3] Implement or unship word-break:break-word

### ISSUE THREAD

**Issue by frivoal** [w3c/csswg-drafts#2390]
We [resolved](https://lists.w3.org/Archives/Public/www-style/2016Mar/0352.html) to add `word-break:break-word` **if** Firefox and Edge found it necessary to implement for web compat reasons.

It's been two years, and it is still not implemented.

If it is needed for compat, it should be implemented and specified. If it is not, it should be removed from the webkit family of browsers.

Since it looks like we're not rushing towards implementation, could we instead have safari and chrome try to remove it? The fact that Edge and Firefox can live without is decently strong evidence of removability.

**Comment by CShepartd** [comment]
You are fighting with web developers who needs "word-break: break-word". For simple example https://jsfiddle.net/ofgd83um/1/
We are not bad deamons, we need to have rule to break word in only neccesary with not breaking size container or all words. Do you think in Chrome supporting it without any reason? (most popular browser, over 57% market share and growing because they providing what developers needs - http://gs.statcounter.com/) We as front-end web-devs need that. You are doing bad thing with fighting against "word-break: break-word" because finally you fighting against us - web developers

**Comment by CShepartd** [comment]
This is BIG mistake and should be reverted

**Comment by astearns** [comment]
We have deferred this, not removed it

**Comment by CShepartd** [comment]
I've seen a mny of sites that are broken in Firefox/IE because they unsupporting this. They have a pattern where they use:

word-break: break-all;
word-break: break-word;

So we ignore the second (unsupported) value and use the first, resulting in undesirable breaks, while Chrome uses the second value and it all looks good.

**Comment by CShepartd** [comment]
>Are there other cases where this is useful?

And this is showing the main problem - you don't know what is "word-break: break-word". Another example? Of course, i have many of this - https://jsfiddle.net/ofgd83um/18/

**Comment by frivoal** [comment]
> And this is showing the main problem - you don't know  [...]

You are under the impression that I am personally trying to kill this. I am not. There are many things that I don't know. But my personal lack of knowledge is not the cause of this.

Every single time this has been discussed in the CSS-WG, the **only** justification that was brought up was compatibility. Based on that, the working group (not me) decided that we should specify it **only if** browsers that currently do not implement it are forced to do so due to compatibility bugs. An incomplete definition was later added to the specification by mistake, against the decision of the group, and then removed when noticed.

This remains the status to this day. I was merely following up on an existing group decision, by saying that implementations and specs should match, so it needs either to be added to all implementations and to the spec OR removed from the implementations that have it.

You are arguing that `word-break:break-word` should exist for other reasons than just compatibility. I am not saying that you are wrong. I am saying that if you want to convince the group to add this property to the specification, the most useful thing to do is not to be angry at me, but to explain in detail all the cases where you think this is a desirable property.

**Comment by dbaron** [comment]
The other key point that needs to be explained is why word-break: break-word is useful in *addition* to [`overflow-wrap: break-word`](https://drafts.csswg.org/css-text-3/#propdef-overflow-wrap).  That requires not just linking to an example but explaining (say, in 3-5 sentences) why the example shows the need.

**Comment by CShepartd** [comment]
@dbaron 
Examples should always better describe problem (and showing real problem in real usage) but if you want usefull addiction to `overflow-wrap: break-word` then it is computing min-content inline size as other values (what is showed in examples above)

**Comment by frivoal** [comment]
> Examples should always better describe problem

Examples illustrate the problem, and are great together with a explanation to make sure there is no misunderstanding, but they are rarely enough.

> I just don't see real reason to deffer it.

Defer means no work. Define now means work. Deferring is the normal answer until enough people are convinced it is needed, either for compatibility reasons or because it is useful.

You are trying to convince a large group of people to do work (write the spec, write the tests, write the code, forever maintain them and take them into account when designing new features). Spending time explaining why that matters is usually more effective than hoping everyone will independently come to the same conclusion.

> [in addition] to overflow-wrap: break-word, it is computing min-content inline size as [if `word-break` was `break-all`].

I understand that this is what it does. I was wondering if you could explain why that is important. Not because I do not believe you, but because explaining why is this best way to help convince the group that this is important enough to warrant the effort, and to evaluate the solution. I can come up with an explanation, but I would prefer to hear from you why you think it is important, instead of guessing your motivations.

You may want to have a look at this description of [the best way to propose features](https://whatwg.org/faq#adding-new-features). It was written for a different working group, but is generally applicable here. It is not a formal process that we have to follow, but it is good advice in terms of convincing a web standards working group to solve a problem you have. "please add this feature" is step 8.

In order to keep CSS maintainable, even when convinced that a problem is worth solving, the group usually wants to consider whether a particular approach is the best way to solve the problem, or whether some other solution would be better.

For example, to solve this problem, I can think of several solutions (not saying they are equally good):
1. `word-break: break-word` as currently implemented in chrome
2. a new value to `overflow-wrap` that makes it affect intrinsic sizing. e.g. `overflow-wrap: break-word` vs `overflow-wrap: break-word allow-shrink`.
3. a new value to `white-space` that changes how intrinsic sizing is calculated. e.g. `white-space: normal` vs `white-space: normal allow-shrink`.
4. A new property to directly affect intrinsic sizing. e.g. `min-content-width: 0`
5. A new table-specific property. e.g. `table-cell-wrap: auto | aggressive`

Solution 1 has the advantage of already existing in some browsers and in some content. It has the disadvantage of making a confusingly named set of properties and values even more confusing, and of lacking the additional flexibility offered by solution 2, 3 or 4. Maybe it matters, maybe it does not.

Solution 2 have the advantage that it can be combined with other values of the `word-break` property, which may be better for internationalization purposes. Since the other values of the `word-break` property are primarily about supporting good typography in Chinese, Japanese and Korean, this may be a good thing. Korean authors may want to use it together with `word-break: keep-all`, Japanese or Chinese authors may want to use it together with `word-break: break-all`.

Solution 3 is similar to solution 2. An additional advantage is that you can use it without setting `overflow-wrap: break-word` if you are interested in the sizing effect but not the wrapping effect and prefer to overflow. A disadvantage is that you need to set more properties if you do want both effects, and that it would not work in table cells without setting both, since table cells cannot overflow.

Solution 4, like solution 3, has the advantage that you can get the sizing effect without the wrapping effect if you want to and to work with all the values of `word-break`, and the same disadvantage of not working in table cells unless you also set has `overflow-wrap: break-word`. It also has the advantage that it can be used in many other different contexts, even when not related to text-wrapping: for example, you could use it to change the intrinsic size of images. This however has the disadvantage that it sets the minimum size to 0, instead of to the widest letter, as `word-break:break-word` would do. Maybe this matters, maybe it doesn't. Maybe making this generic is great, maybe it is overkill.

Solution 5 may be better if the actual problem is limited to text within tables, and if we don't want to make a generic solution to a narrow problem.

But listing solutions is  suggested as "step 5" in the above process, because getting a list of detailed requirements is very helpful to chose between the various possible solutions.

**Comment by CShepartd** [comment]
>Defer means no work. Define now means work

**When something working in more than 83% of web browsers in the world i think it shouldn't be deferred** (Chrome, Opera, Safari, UC-Browser, Vivaldi have working `word-break:break-word`). And it will grow because in Edge issue tickets about missing `word-break:break-word` they responsing about adding this soon. Web browsers usage stats (http://gs.statcounter.com/) are without emotion and showing the best your misunderstood of situation with `word-break:break-word`

Your "solutions" are wrong too because not working in opposite to working `word-break: break-word`. If you dont see why it not working see screen and focus on width (body max is 400, live exaple there https://jsfiddle.net/ofgd83um/32/)
![defferingwaswrong](https://user-images.githubusercontent.com/24432753/38867478-79a67f54-4244-11e8-8094-26143028964d.png)

In summary - `word-break: break-word` working in more than 83% of web browsers (by usage) and shouldn't be deferred

**Comment by frivoal** [comment]
> And it will grow because in Edge issue tickets about missing word-break:break-word they responsing about adding this soon

The existing resolution, which still stands, is to add it to the spec as soon as Edge or firefox does so. If they do so, we will add it to the spec. If they find that it is not necessary to do so, I will trust their analysis more than a blunt percentage, we will not add it to the spec, unless we find other good reasons.

If you think they will add it soon, you should not be worried, because we will add it to the spec as well.

If you think they will not add it, and you would like this feature anyway, I respectfully suggest that you spend your energy convincing people that it is an important feature by following the suggestions dbaron and I have given you, or by following the [guide to proposing features](https://whatwg.org/faq#adding-new-features).

If you are not interested in providing constructive feedback, and only want to complain and belittle the few people who are listening to you, I suggest you stop discussing this entirely.

**Comment by CShepartd** [comment]
@frivoal You are talking the same trash talk everytime when you dont have any contarguments. When I showed your misunderstood of `word-break:break-word` you are pretending nothing happend. When showing real stats of usage you are escaping again to same trash talk. You are on big mistake in a W3C cuz of your inconpetence and special ignorance

**Comment by astearns** [comment]
That's not acceptable @CShepartd. I'm removing the comment above. If you continue to post abusively (here or in IRC) I'll ban you.

**Comment by fantasai** [comment]
Note that the behavior of this feature&mdash;allowing breaks anywhere if the word cannot fit its container (per `word-wrap/overflow-wrap: break-word`) and <em>also</em> having these break opportunities affect the min-content size&mdash;is now specified for `word-wrap/overflow-wrap: break-word`.  See https://github.com/w3c/csswg-drafts/issues/2682

This means that once implementations catch up, there is no use case requiring `word-break: break-word` to be added (since `overflow-wrap: break-word` now has this behavior), and the only reason to add this second syntax would be if it's needed for Web-compat.

**Comment by CShepartd** [comment]
@fantasai 
It will pass test? https://jsfiddle.net/ofgd83um/53/

**Comment by emilio** [comment]
Per https://bugzilla.mozilla.org/show_bug.cgi?id=1296042#c24 (and co.) Blink is not planning to unship this anytime soon, and sites not using `overflow-wrap: anywhere` are still giving Gecko compat headache... It'd be good to figure out what to do here, even if it's "putting it on the compat spec".

CC @miketaylr

**Comment by CShepartd** [comment]
Finally. Already implement in Gecko (Firefox 68 Nightly)

**Comment by john-cj** [comment]
> Note that the behavior of this feature—allowing breaks anywhere if the word cannot fit its container (per `word-wrap/overflow-wrap: break-word`) and _also_ having these break opportunities affect the min-content size—is now specified for `word-wrap/overflow-wrap: break-word`. See #2682
> 
> This means that once implementations catch up, there is no use case requiring `word-break: break-word` to be added (since `overflow-wrap: break-word` now has this behavior), and the only reason to add this second syntax would be if it's needed for Web-compat.

@fantasai Does it mean that `word-break: break-word` is now redundant and we can simply replace it with `overflow-wrap: break-word` without changing any other things?

And does it mean that the following statement by MDN is no longer correct?

> Note: In contrast to word-break, overflow-wrap will only create a break if an entire word cannot be placed on its own line without overflowing. (Source: https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap)

I ask from a perspective of an average web developer.

**Comment by frivoal** [comment]
> Does it mean that `word-break: break-word` is now redundant and we can simply replace it with `overflow-wrap: break-word` without changing any other things?

Yes.

> And does it mean that the following statement by MDN is no longer relevant?

> > Note: In contrast to word-break, overflow-wrap will only create a break if an entire word cannot be placed on its own line without overflowing. (Source: https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap)

For other values of the `word-break` property, that statement is true, and there's a difference between the two properties. For `word-break: break-word`, it behaves the same as `overflow-wrap: break-word`, and only takes allows breaks "if an entire word cannot be placed on its own line without overflowing".
